### ISSUE TITLE
wlinknet upgrade: excessive sync retries on ./bin/server.  Excessive error notices. False prompt to delete data directory on sync crash. Almost guaranteed crashing (on sync, not normal use).   Sync needs manual restarts and lots of patience (20+ hours) to sync without bootstrap.

### ISSUE THREAD

**Issue by orchardstreet** [mr-pandabear/panda-coin#126]
This is a bug report for the wlinknet upgrade branch of the main repo.  I tested using that branch, also known as the 'new node'.

The syncing has been improved since last few months.  In that a bootstrap is not needed if you have patience and many hours to spare.  This is very good.   Longstanding issues with the nodes were also fixed by wlinknet, which is great.  Random crashing (after sync) seems to be gone, meaning exchanges can probably take PDN out of maintenance (with a bootstrap)

However, even with the upgrade, the syncing will exceed max retries, and need manual restarts.  After it exceeds max retries after crashing, it will prompt this user to delete the data directory.  They should not delete the data directory.   As it will continue syncing if restarted at this point.  A bandaid solution would be to suppress the log errors, and delete the string telling the user to delete the directory folder, with automated restarts of the syncing process.

A long term solution, which wlinknet said he may use, involves building a map of the bad blocks so to navigate around them.  Erik also said a JSON file fix may fix most or all of this issue.

tl;dr syncing without a bootstrap works on the upgrade, but it is not reasonable to ask users or exchanges to do so considering the time, excessive warning, and prompts to delete data directory.  In my opinion, more work is needed on this before release a new node, or at least a bandaid solution.   As mentioned earlier, wlinknet's upgrade seems to solve many issues and is very good, but I think it's worth noting issues before release, so as to fix or to use a band aid.   Thanks.

**Comment by orchardstreet** [comment]
would like to again note that everything listed above is probably an improvement to the state of the master branch.  And that wlinknet's work is an overall net positive, and people are very grateful for his work.  Just wanted to keep people updated with how the upgrade is looking.

**Comment by orchardstreet** [comment]
The initial point of his upgrade seems to be to prevent nodes from crashing, which so far seems to be accomplished and alone is worth upgrading over in my humble opinion.  Most node operators are probably already running his branch and his branch is being used as the main node software already I think.
